controllers/maintenance_mode_checker_test.go

```diff
@@ -62,88 +65,290 @@ var _ = Describe("maintenance_mode_checker", func() {
-						continue
-					}
+				Expect(adminClient.SetProcessesUnderMaintenance([]fdbv1beta2.ProcessGroupID{targetProcessGroup}, time.Now().Add(-1*time.Minute).Unix())).NotTo(HaveOccurred())
+				Expect(adminClient.SetProcessesUnderMaintenance([]fdbv1beta2.ProcessGroupID{secondStorageProcess}, time.Now().Add(-1*time.Minute).Unix())).NotTo(HaveOccurred())
+			})
 
-					processGroup.UpdateCondition(fdbv1beta2.MissingProcesses, true)
-				}
+			It("should requeue", func() {
+				Expect(requeue).NotTo(BeNil())
+				Expect(requeue.delayedRequeue).To(BeTrue())
+				Expect(requeue.message).To(Equal("Waiting for 1 processes in zone operator-test-1-storage-1 to be updated"))
 			})
 
-			It("should be requeued", func() {
-				Expect(requeue).ToNot(BeNil())
-				Expect(requeue.message).To(Equal("Waiting for 1 process groups in zone operator-test-1-storage-1 to be updated"))
+			It("shouldn't reset the maintenance", func() {
+				maintenanceZone, err := adminClient.GetMaintenanceZone()
+				Expect(err).NotTo(HaveOccurred())
+				Expect(maintenanceZone).To(Equal(string(currentMaintenanceZone)))
+			})
+
+			It("shouldn't remove the entry from the maintenance list for the second entry", func() {
+				processesUnderMaintenance, err := adminClient.GetProcessesUnderMaintenance()
+				Expect(err).NotTo(HaveOccurred())
+				Expect(processesUnderMaintenance).To(HaveLen(1))
+				Expect(processesUnderMaintenance).To(HaveKey(secondStorageProcess))
 			})
 		})
 
-		When("all Pods are updated", func() {
-			It("should not be requeued", func() {
+		When("one processes for a different zone is in the maintenance list and was recently added and the process is missing", func() {
+			BeforeEach(func() {
+				// The maintenance is targeted for the future.
+				Expect(adminClient.SetProcessesUnderMaintenance([]fdbv1beta2.ProcessGroupID{secondStorageProcess}, time.Now().Add(1*time.Minute).Unix())).NotTo(HaveOccurred())
+				adminClient.MockMissingProcessGroup(secondStorageProcess, true)
+			})
+
+			It("should requeue", func() {
+				Expect(requeue).NotTo(BeNil())
+				Expect(requeue.delayedRequeue).To(BeTrue())
+				Expect(requeue.message).To(Equal("Waiting for 1 processes in zone operator-test-1-storage-1 to be updated"))
+			})
+
+			It("shouldn't reset the maintenance", func() {
+				maintenanceZone, err := adminClient.GetMaintenanceZone()
+				Expect(err).NotTo(HaveOccurred())
+				Expect(maintenanceZone).To(Equal(string(currentMaintenanceZone)))
+			})
+
+			It("shouldn't remove the entry from the maintenance list", func() {
+				processesUnderMaintenance, err := adminClient.GetProcessesUnderMaintenance()
+				Expect(err).NotTo(HaveOccurred())
+				Expect(processesUnderMaintenance).To(HaveLen(1))
+				Expect(processesUnderMaintenance).To(HaveKey(secondStorageProcess))
+			})
+		})
+
+		When("one processes for a different zone is in the maintenance list and was added multiple minutes ago", func() {
```

Comment:
I'm a bit confused about the logic for this case

Reply:
This reflects that case where the process is long enough in the list to not block the removal of the maintenance zone, but not long enough to be considered stale. I update the when clause.